tensorflow/lite/micro/kernels/xtensa/unidirectional_sequence_lstm.cc

```diff
@@ -670,6 +670,11 @@ TfLiteStatus PrecomputeZeroPointTimesWeightWithBias(
   TF_LITE_ENSURE_EQ(context, weight_shape.DimensionsCount(), 2);
   const int row = weight_shape.Dims(0);
   const int col = weight_shape.Dims(1);
+  /* Releases ownership of its stored pointer, by returning its value
+     and replacing it with a null pointer. */
+  int *temp = output->release();
```

Comment:
I am not entirely sure why this is required.

Reply:
Hi @deqiangc,

In the case of "new", it allocates memory and calls the constructors to initialize the memory and the pointer will be set to nullptr.
But TFLM uses AllocatePersistentBuffer/AllocateScratchBuffer so the pointer will not be set to nullptr, instead, it uses some random values. Here the behavior of reset is undefined.

To resolve the issue, we are calling release before reset. So it does initializes the pointer to nullptr then reset will work as expected.

Thanks,
Bhanu